com.unity.perception/Editor/Randomization/VisualElements/Parameter/ParameterElement.cs

```diff
@@ -87,15 +87,30 @@ void BindItem(VisualElement e, int i)
                     // First delete sets option to null, second delete removes option
                     var numOptions = optionsProperty.arraySize;
                     optionsProperty.DeleteArrayElementAtIndex(i);
+
                     if (numOptions == optionsProperty.arraySize)
+                    {
                         optionsProperty.DeleteArrayElementAtIndex(i);
+                    }
+
+                    ResetProbabilities();
```

Comment:
I think the reset line should only be called if `uniform` is enabled, right? Otherwise we keep the probabilities as they are.

Reply:
I'm not sure on that, unless we default all new entries to 0% probability. If someone has their probabilities custom set and then they add in two new parameters then, either the new ones have to be set to 0 out of the box, or we need to rebalance. I'm not sure what the right answer is here, or if there is one. Thoughts?

Reply:
I see your point. My thoughts are, if the user has manually set probabilities, they are aware of the feature and how to use it. So when they add new options they would probably know they need to set probabilities for them. 
On the other hand, resetting the probabilities can potentially remove numbers that the user has carefully set.

Reply:
The default behaviour when uniform is disabled and a single option is added is also to set the probability to 0 right?

Reply:
I'll have to get a clean branch to verify but I believe it recalculates

Reply:
So, the current behaviour is a bit inconsistent depending on whether there are previous options already added, and whether a single option or a folder is added next. I think what we could when uniform is **disabled**:
- When a folder is added, all the new options get a probability equal to 1/(number of new options). This is how it works now. It does lead to the sum of all options being more than 1, but that's ok, it is normalized later.
- When a single option is added, we set its probability to 0. Ideally, whenever an option has a 0 probability we could also show a warning box saying there are items with 0 probability. The current behaviour is to set this to the probability of the last element previously in the list, or 0 if the list is empty.

I am mainly concerned about changing probabilities that are already set by the user. That would not be a good UX.

Reply:
Ok, reworked it so that added entries default to probability of 0 if uniform is unchecked